system/Helpers/filesystem_helper.php

```diff
@@ -218,13 +218,7 @@ function get_filenames(
                 }
 
                 if ($includeDir || ! $object->isDir()) {
-                    if ($includePath === false) {
-                        $files[] = $basename;
-                    } elseif ($includePath === null) {
-                        $files[] = str_replace($sourceDir, '', $name);
-                    } else {
-                        $files[] = $name;
-                    }
+                    $files[] = $includePath === false ? $basename : ($includePath === null ? str_replace($sourceDir, '', $name) : $name);
```

Comment:
This is more difficult to read to me.

Reply:
Benefit one liner is `reduce` byte files. I think three condition to one liner readable for me

Reply:
Shorter code does not make sense. If it is good, we must use single letter variable names like `$a`.
> Code should be written to minimize the time it would take for someone else to understand it.

The benchmark code should be:
```php
        helper('filesystem');

        $tempAverage = [];

        for ($i = 0; $i < 10; $i++) {
            timer('get_filenames'.$i); // must use unique name
            get_filenames(ROOTPATH, true, true, true);
            timer('get_filenames'.$i);

            $tempAverage[$i] = timer()->getElapsedTime('get_filenames'.$i);
        }

        echo array_sum($tempAverage) / count($tempAverage);
```

But the true difference is like this:
```php
        $iterator = new \CodeIgniter\Debug\Iterator();

        $iterator->add(
            'before',
            static function ($includePath = true, $basename = '', $sourceDir = '', $name = '') {
                if ($includePath === false) {
                    $files[] = $basename;
                } elseif ($includePath === null) {
                    $files[] = str_replace($sourceDir, '', $name);
                } else {
                    $files[] = $name;
                }
            }
        );

        $iterator->add(
            'after',
            static function ($includePath = true, $basename = '', $sourceDir = '', $name = '') {
                $files[] = $includePath === false ? $basename : ($includePath === null ? str_replace($sourceDir, '', $name) : $name);
            }
        );

        return $iterator->run(100000);
```
macOS:
<table>
			<thead>
				<tr>
					<td>Test</td>
					<td>Time</td>
					<td>Memory</td>
				</tr>
			</thead>
			<tbody>
				<tr>
				<td>before</td>
				<td>0.2164</td>
				<td>0 Bytes</td>
			</tr><tr>
				<td>after</td>
				<td>0.2257</td>
				<td>0 Bytes</td>
			</tr>
			</tbody>
		</table>

Reply:
Revert change in Filesystem

Reply:
Thank you. Please remove other changes.